### PR TITLE
prefer esperanto to es6-module-transpiler

### PR DESCRIPTION
**Pull request by stefanpenner**
- we get better spec compat (reasonable cycles work)
- it’s extremely fast

```
node ./benchmarks/modules-source-maps.js ./test-input/controller.js
running...
  - traceur x 360 ops/sec ±5.57% (74 runs sampled)
  - 6to5 (AMD) x 63.98 ops/sec ±5.54% (59 runs sampled)
  - 6to5 (CJS) x 61.93 ops/sec ±7.14% (54 runs sampled)
  - esperanto (AMD) x 197 ops/sec ±11.29% (66 runs sampled)
  - esperanto (CJS)  x 349 ops/sec ±5.76% (73 runs sampled)

node ./benchmarks/modules.js ./test-input/controller.js
running...
  - traceur x 297 ops/sec ±6.98% (71 runs sampled)
  - 6to5 (AMD) x 20.99 ops/sec ±6.84% (40 runs sampled)
  - 6to5 (CJS) x 19.26 ops/sec ±12.74% (37 runs sampled)
  - esperanto (AMD) x 191 ops/sec ±20.73% (51 runs sampled)
  - esperanto (CJS)  x 898 ops/sec ±19.33% (51 runs sampled)
```

@eventualbuddha has been directing his energy there lately.

cc @rich-harris

- [ ] enable source maps (@ef4 does your concat stuff handle inline sourcemaps?)

### CODE REVIEW

**Comment by rwjblue** [comment]
@stefanpenner - Have you reviewed ES3 compat?

**Comment by stefanpenner** [comment]
the output is es3 safe

**Comment by stefanpenner** [comment]
this does remove the `patchRelativeImports`, if we think we need it we can push it down into esperanto.

**Comment by Rich-Harris** [comment]
I just raised https://github.com/esperantojs/esperanto/issues/58 for the `patchRelativeImports` stuff

**Comment by stefanpenner** [comment]
@Rich-Harris :+1:

**Comment by ef4** [comment]
Only concern is that I was about to revisit this code to fix https://github.com/ember-cli/ember-cli/issues/2949.

It looks like esperanto leaves missing module references as external dependencies, which is good. But does it expose that list to us so we can go and find them in other paths and package them into vendor?

**Comment by Rich-Harris** [comment]
@ef4 good thinking - have raised https://github.com/esperantojs/esperanto/issues/59

File: package.json [hunk 1]
```diff
@@ -10,10 +10,15 @@
...
+    "broccoli": "^0.13.3",
+    "broccoli-caching-writer": "^0.5.3",
     "broccoli-kitchen-sink-helpers": "^0.2.5",
+    "chai": "^1.10.0",
     "es6-module-transpiler": "^0.3.6",
+    "esperanto": "^0.6.3",
     "mkdirp": "^0.5.0",
+    "mocha": "^2.1.0",
+    "rsvp": "^3.0.16",
     "walk-sync": "^0.1.3"
```
Comment (ef4) [inline package.json]: Any reason you moved all these devDependencies into here?

Comment (stefanpenner) [inline package.json]: npm orders them alphabetically  automatically when using the CLI

Comment (ef4) [inline package.json]: I'm not referring to the order. They are already listed as `devDependencies`, and they got added to `dependencies`.

Comment (stefanpenner) [inline package.json]: hmm strange, let me fix

**Comment by stefanpenner** [comment]
also fixing tests.

@ef4 wanna hook up travis?

**Comment by ef4** [comment]
Travis enabled.

**Comment by stefanpenner** [comment]
@ef4 its not the end of the world if this PR sits pending until esperanto does what you need. I don't want to block you

**Comment by jclem** [comment]
Is this in a release? I still see undefined modules in cyclic dependencies, as described in ember-cli/ember-cli#1009.

**Comment by rwjblue** [comment]
Yes, it has been in a few releases now...

**Comment by jclem** [comment]
Actually I'm on 0.5.1 of this package and am still having this problem.

**Pull request merged**